.github/workflows/tests/BUILD

```
@@ -0,0 +1,28 @@
+# Copyright 2023 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+# This contains various smoke tests for out GitHub Actions workflows.
+# These aren't meant to be comprehensive, as they can be quite expensive to run. However, they are
+# still useful as they can expose bugs before the action is run in "prod".
+
+
+shell_sources(name="tests")
+
+
+def _run_test(filename):
+    filepath = build_file_dir() / filename
+    return f"chmod +x {filepath} && {filepath}"
```

Just a minor thing from https://github.com/pantsbuild/pants/pull/19230/files/888c8d380f1edb5c50c4e71cf8f287ec41f961a4#r1223619489

This could avoid the `chmod +x` fiddle using something like:

```suggestion
def _run_test(filename):
    filepath = build_file_dir() / filename
    return f"bash {filepath}"
```

And, at that point, it could probably just be inlined below `command=f"bash {build_file_dir()}/smoke_test_auth_cherry_picker.sh"`?